### PR TITLE
python3.pkgs.django-webpush: init at 0.3.4

### PR DESCRIPTION
**Pull request by DerDennisOP**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/django-webpush/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  doCheck = false;
+
+  pythonImportsCheck = [
+    "webpush"
+  ];
+
+  meta = with lib; {
+    description = "Django-Webpush is a Package made for integrating and sending Web Push Notification in Django Application.";
+    homepage = "https://github.com/safwanrahman/django-webpush/";
+    changelog = "https://github.com/safwanrahman/django-webpush/releases/tag/${version}";
```
Comment (mbalatsko) [inline pkgs/development/python-modules/django-webpush/default.nix:50]:
```suggestion
    changelog = "https://github.com/safwanrahman/django-webpush/releases/tag/${src.rev}";
```

File: pkgs/development/python-modules/django-webpush/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  src = fetchFromGitHub {
+    owner = "safwanrahman";
+    repo = "django-webpush";
+    rev = "refs/tags/${version}";
+    hash = "sha256-Mwp53apdPpBcn7VfDbyDlvLAVAG65UUBhT0w9OKjKbU=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
```
Comment (mbalatsko) [inline pkgs/development/python-modules/django-webpush/default.nix:27]: Can we do it using `pythonRelaxDeps`? https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonrelaxdepshook-using-pythonrelaxdepshook

File: pkgs/development/python-modules/django-webpush/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{ lib
+, buildPythonPackage
+, django
+, fetchFromGitHub
+, pytest-django
```
Comment (mbalatsko) [inline pkgs/development/python-modules/django-webpush/default.nix:5]: Ununsed
```suggestion
```

File: pkgs/development/python-modules/django-webpush/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+, python
+, pythonOlder
+, pywebpush
+, setuptools-scm
+}:
+
+buildPythonPackage rec {
+  pname = "django-webpush";
+  version = "0.3.4";
+  format = "setuptools";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-webpush/default.nix:15]:
```suggestion
  pyproject = true;
```
We have a new parameter for the format.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/development/python-modules/django-webpush/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  # nothing to test
+  doCheck = false;
+
+  pythonImportsCheck = [
+    "webpush"
+  ];
+
+  meta = with lib; {
+    description = "Django-Webpush is a Package made for integrating and sending Web Push Notification in Django Application.";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-webpush/default.nix:48]:
```suggestion
    description = "A Package made for integrating and sending Web Push Notification in Django Application";
```
> Don’t include a period at the end. Don’t include newline characters. Capitalise the first character. For brevity, don’t repeat the name of package — just describe what it does.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/development/python-modules/django-webpush/default.nix [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    homepage = "https://github.com/safwanrahman/django-webpush/";
+    changelog = "https://github.com/safwanrahman/django-webpush/releases/tag/${src.rev}";
+    license = licenses.gpl3;
```
Comment (natsukium) [inline pkgs/development/python-modules/django-webpush/default.nix:51]:
```suggestion
    license = licenses.gpl3Plus;
```
https://github.com/safwanrahman/django-webpush/blob/0561306f133619297d7d66dd683690edf55cf325/README.md?plain=1#L211

**Review by natsukium — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/django-webpush/default.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildPythonPackage
+, django
+, fetchFromGitHub
+, python
```
Comment (natsukium) [inline pkgs/development/python-modules/django-webpush/default.nix:5]:
```suggestion
```
Please remove an unused input.

**Review by natsukium — COMMENTED** [review]
Besides this, LGTM.

**Comment by natsukium** [comment]
@ofborg build python311Packages.django-webpush python310Packages.django-webpush

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 258404` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.django-webpush</li>
    <li>python310Packages.django-webpush.dist</li>
    <li>python311Packages.django-webpush</li>
    <li>python311Packages.django-webpush.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Comment by DerDennisOP** [comment]
@natsukium could you also lookt at my other commits :) pls

**Pull request merged**